File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/question_0164_Maximum_Gap.java
package com.lanceolata.coding.algorithms.java.leetcode;

import java.util.Arrays;

public class question_0164_Maximum_Gap {
    public int maximumGap(int[] nums) {
        if (nums == null || nums.length < 2) {
            return 0;
        }
        Arrays.sort(nums);
        int res = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            res = Math.max(res, nums[i + 1] - nums[i]);
        }
        return res;
    }
}

File: algorithms/algorithms-cplusplus/leetcode/question_0164_Maximum_Gap.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int maximumGap(vector<int>& nums) {
    if (nums.size() < 2) {
      return 0;
    }
    sort(nums.begin(), nums.end());
    int res = 0;
    for (int i = 0; i < nums.size() - 1; i++) {
      res = max(res, nums[i + 1] - nums[i]);
    }
    return res;
  }
};
